File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.py
class Solution:
    def minDeletions(self, s: str) -> int:
        
        count = [0]*26
        
        for c in s:
            count[ord(c) - ord("a")] += 1
            
        
        count.sort(reverse=True)
        flag = count[0]
        ans = 0
        
        for val in count:
            if val > flag:
                if flag>0:
                    ans += val - flag
                else:
                    ans += val
            flag = min(flag-1, val-1)
            
        return ans

File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.cpp
class Solution {
public:
    int minDeletions(string s) {
        
        vector<int> v(26, 0);
        for(auto c : s){
            v[c-'a']++;
        }
        
        sort(v.begin(), v.end(), greater<int>());
        
        int maxlen = v[0];
        int ans=0;
        
        for(auto val : v){
            
            if(val > maxlen){
                
                if(maxlen > 0){
                    ans += val - maxlen;
                }
                else{
                    ans += val;
                }
            }
            
            maxlen = min(maxlen-1, val-1);
        }
        
        return ans;
    }
};